ckanclient/__init__.py

```diff
@@ -0,0 +1,165 @@
+        self.lfs_url = lfs_url
+        self.auth = CkanAuthApi(api_url, api_key, organization, dataset_id)
+
+    def action(
+        self,
+        name,
+        payload,
+        http_get=False,
+        transform_payload=None,
+        transform_response=None,
+    ):
+        """Gives you direct access to the CKAN Action API
+        (https://docs.ckan.org/en/2.8/api/).
+
+        Attributes:
+            name (str): The action name, e.g. site_read, package_show…
+            payload (dict): The payload being sent to CKAN. If a payload is
+                provided for a GET request, it will be converted to URL
+                parameters and each key will be converted to snake case.
+            http_get (bool): Optional, if `True` will make `GET` request,
+                otherwise `POST`.
+            transform_payload (func): Function to mutate the `payload` before
+                making the request (useful to convert to and from CKAN and
+                Frictionless formats).
+            transform_response (func): Function to mutate the response data
+                before returning it (useful to convert to and from CKAN and
+                Frictionless formats).
+
+        Returns:
+            dict: the API response converted from JSON to a Python dictionary.
+        """
+        url = f'{self.api_url}/api/3/action/{name}'
+        headers = {
+            'Content-Type': 'application/json;charset=utf-8',
+            'Authorization': self.api_key,
+        }
+        if transform_payload:
+            payload = transform_payload(payload)
+
+        if http_get:
+            params = {camel_to_snake(k): v for k, v in payload.items()}
+            response = requests.get(url, headers=headers, params=params)
+        else:
+            response = requests.post(url, headers=headers, data=payload)
+
+        if response.status_code < 200 or response.status_code >= 300:
+            msg = (
+                'CKAN Action API returned HTTP Status Code '
+                f'{response.status_code}:\n{response.text}'
+            )
+            raise CkanClientError(msg)
+
+        data = response.json()
+        return transform_response(data) if transform_response else data
+
+    def create(self, dataset_name_or_metadata):
+        """Creates a new dataset.
+
+        Attributes:
+            dataset_name_or_metadata (str or dict): It is either a string
+                being a valid dataset name or metadata for the dataset in
+                Frictionless format.
+
+        Returns:
+            dict: the newly created dataset.
+        """
+        metadata = dataset_name_or_metadata
+        if isinstance(metadata, str):
+            metadata = {
+                'name': dataset_name_or_metadata,
+                'owner_org': self.organization,
+            }
+
+        return self.action(
+            'package_create',
+            metadata,
+            transform_payload=ckan_to_frictionless.dataset,
```

Comment:
As an example i would do the transform here ... rather than put this into the action api.